### PR TITLE
LG-4138 | LG-4150: [CCP] Select updates

### PR DESCRIPTION
**Pull request by stephl3**
## ✍️ Proposed changes

- updates styling
- adds default `errorMessage`
- adds `valid` state with `successMessage` prop

🎟 _Jira ticket:_ [LG-4138](https://jira.mongodb.org/browse/LG-4138)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

## 🧪 How to test changes

Figma: https://www.figma.com/file/WmbvStSle8GtLg5RqEt2oh/Component-Consistency-Project?type=design&node-id=59%3A15314&mode=design&t=Wf5VD9qqs9EmPGXy-1
Review storybook and chromatic

### CODE REVIEW

File: packages/select/src/MenuButton/MenuButton.styles.ts [hunk 1]
```diff
@@ -103,10 +103,10 @@ export const menuButtonFocusStyle: Record<Theme, string> = {
 
 export const menuButtonDeselectedStyles: Record<Theme, string> = {
   [Theme.Light]: css`
-    color: ${palette.gray.dark1};
+    color: ${palette.gray.base};
```
Comment (bruugey) [inline packages/select/src/MenuButton/MenuButton.styles.ts:106]: tokens

File: packages/select/src/MenuButton/MenuButton.styles.ts [hunk 2]
```diff
@@ -103,10 +103,10 @@ export const menuButtonFocusStyle: Record<Theme, string> = {
...
   `,
   [Theme.Dark]: css`
-    color: ${palette.gray.light1};
+    color: ${palette.gray.dark1};
```
Comment (bruugey) [inline packages/select/src/MenuButton/MenuButton.styles.ts:109]: tokens

File: packages/select/src/MenuButton/MenuButton.styles.ts [hunk 3]
```diff
@@ -132,8 +132,14 @@ export const menuButtonDisabledThemeStyles: Record<Theme, string> = {
     css`
       &[aria-disabled='true'] {
         background-color: ${palette.gray.light2};
+        border-color: ${palette.gray.light1};
```
Comment (bruugey) [inline packages/select/src/MenuButton/MenuButton.styles.ts:135]: ditto

File: packages/select/src/MenuButton/MenuButton.styles.ts [hunk 4]
```diff
@@ -201,3 +212,29 @@ export const menuButtonErrorStyle: Record<Theme, string> = {
     }
   `,
 };
+
+export const validColor: Record<Theme, string> = {
+  [Theme.Light]: palette.green.dark1,
```
Comment (bruugey) [inline packages/select/src/MenuButton/MenuButton.styles.ts:217]: tokens

File: packages/select/src/MenuButton/MenuButton.styles.ts [hunk 5]
```diff
@@ -201,3 +212,29 @@ export const menuButtonErrorStyle: Record<Theme, string> = {
...
+  [Theme.Dark]: palette.green.base,
+};
+
+export const menuButtonValidStyle: Record<Theme, string> = {
+  [Theme.Light]: css`
+    border-color: ${errorColor[Theme.Light]};
```
Comment (bruugey) [inline packages/select/src/MenuButton/MenuButton.styles.ts:223]: this feels verbose, no? we're already in theme.light i think you can just use our tokens here

File: packages/select/src/Select/Select.tsx [hunk 6]
```diff
@@ -615,29 +613,13 @@ export const Select = forwardRef<HTMLDivElement, SelectProps>(
...
-                    // so it doesn't peek around the menu corner
-                    color: transparent;
-                  `]: open,
-                },
-              )}
-            >
-              {errorMessage}
-            </span>
-          )}
+          <StateFeedback
```
Comment (bruugey) [inline packages/select/src/Select/Select.tsx:616]: should this component live in FormField as well? (Does it already, should we just export it?)

File: packages/select/src/StateFeedback/StateFeedback.styles.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  align-items: center;
+  gap: ${spacing[100]}px;
+`;
+
+export const hideContainerStyle = css`
+  opacity: 0;
+`;
+
+export const errorIconThemeStyle: Record<Theme, string> = {
+  [Theme.Light]: palette.red.base,
```
Comment (bruugey) [inline packages/select/src/StateFeedback/StateFeedback.styles.ts:26]: tokens 😛

File: packages/select/src/StateFeedback/StateFeedback.styles.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+export const errorTextMobileStyles = css`
+  ${MobileMediaQuery} {
+    font-size: ${mobileSizeSet.description.text}px;
+    line-height: ${mobileSizeSet.description.lineHeight}px;
+  }
+`;
+
+export const validIconThemeStyle: Record<Theme, string> = {
+  [Theme.Light]: palette.green.dark1,
```
Comment (bruugey) [inline packages/select/src/StateFeedback/StateFeedback.styles.ts:43]: this can be one line in the component itself: 

`color[mode].iconColor.success`

Comment (bruugey) [reply packages/select/src/StateFeedback/StateFeedback.styles.ts:43]: we should look for the opportunities to consolidate these style objects where possible.

File: packages/select/src/StateFeedback/StateFeedback.styles.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  [Theme.Dark]: palette.green.base,
+};
+
+export const validMessageThemeStyle: Record<Theme, string> = {
+  [Theme.Light]: css`
+    color: ${palette.green.dark1};
```
Comment (bruugey) [inline packages/select/src/StateFeedback/StateFeedback.styles.ts:49]: ditto here

File: packages/select/src/StateFeedback/StateFeedback.tsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  const baseFontSize = useUpdatedBaseFontSize();
+
+  const isErrorState = state === State.Error;
+  const isValidState = state === State.Valid;
+  const showStateFeedback = (isErrorState || isValidState) && !disabled;
+
+  const iconProps = {
+    glyph: isErrorState ? 'Warning' : 'Checkmark',
+    fill: isErrorState
+      ? errorIconThemeStyle[theme]
```
Comment (bruugey) [inline packages/select/src/StateFeedback/StateFeedback.tsx:39]: `tokens.color[mode].iconColor.error.default`

File: packages/select/src/MenuButton/MenuButton.tsx [hunk 11]
```diff
@@ -80,9 +76,9 @@ const MenuButton = React.forwardRef<HTMLButtonElement, MenuButtonProps>(
           menuButtonFocusStyle[theme],
           {
             [menuButtonDeselectedStyles[theme]]: deselected,
+            [menuButtonErrorStyle[theme]]: state === State.Error,
```
Comment (TheSonOfThomp) [inline packages/select/src/MenuButton/MenuButton.tsx:79]: just a nit, but I noticed in Combobox you're creating a new var (`const isErrorState = state === State.error`). Should we strive for consistency there? (personally, I don't think it matters, just curious what others thought)

File: packages/select/src/Select/Select.spec.tsx [hunk 12]
```diff
@@ -936,14 +936,6 @@ describe('packages/select', () => {
       expect(isError).toBeTruthy();
       expect(getErrorMessage()).toBeInTheDocument();
     });
-
-    test('error message is not present if "errorMessage" is empty', () => {
```
Comment (TheSonOfThomp) [inline packages/select/src/Select/Select.spec.tsx:940]: Should we test the default error message though?

File: packages/select/src/StateFeedback/StateFeedback.styles.ts (hunk 7 above)
Comment (TheSonOfThomp) [reply packages/select/src/StateFeedback/StateFeedback.styles.ts:26]: 👍

File: packages/select/src/StateFeedback/StateFeedback.tsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+  errorIconThemeStyle,
+  errorTextMobileStyles,
+  hideContainerStyle,
+  validIconThemeStyle,
+  validMessageThemeStyle,
+  validMessageTypeScaleStyles,
+} from './StateFeedback.styles';
+import { StateFeedbackProps } from './StateFeedback.types';
+
+export const StateFeedback = ({
```
Comment (TheSonOfThomp) [inline packages/select/src/StateFeedback/StateFeedback.tsx:22]: Is this unique to Select? Could this move to `FormField?

**Review by TheSonOfThomp — APPROVED** [review]

File: packages/select/src/StateFeedback/StateFeedback.tsx (hunk 13 above)
Comment (stephl3) [reply packages/select/src/StateFeedback/StateFeedback.tsx]: moved :)

File: packages/select/src/Select/Select.spec.tsx (hunk 12 above)
Comment (stephl3) [reply packages/select/src/Select/Select.spec.tsx:940]: I added the default message at the form field level

**Review by TheSonOfThomp — APPROVED** [review]

**Pull request merged**